docs/architecture/application/forms.md

```diff
@@ -127,4 +127,306 @@ def initial_input_form_generator(product: UUIDstr, product_name: str) -> FormGen
     return {**user_input.dict(), **user_input_node.dict()}
 ```
 
+## custom form field
+
+You can create a custom field component in frontend and can take reference from components in `orchestrator-gui/src/lib/uniforms-surfnet/src`.
+
+For it to show up in the form, you have to do 2 things, a pydantic type/class in the backend and add the component to the `AutoFieldLoader.tsx`.
+
+as example I will create a custom field with name field and group select field.
+
+### pydantic type/class in backend
+
+Create a pydantic type/class.
+
+``` python
+from uuid import UUID
+
+
+class ChooseUser(int):
+    group_id: UUID  # type:ignore
+
+    @classmethod
+    def __modify_schema__(cls, field_schema: dict[str, Any]) -> None:
+        uniforms: dict[str, Any] = {}
+
+        if cls.group_id:
+            uniforms["groupId"] = cls.group_id
+
+        field_schema.update(format="ChooseUser", uniforms=uniforms)
+```
+
+And add it to a form:
+
+``` python
+def initial_input_form_generator(product: UUIDstr, product_name: str) -> FormGenerator:
+    class ChoseUserForm(FormPage):
+        class Config:
+            title = product_name
+
+        user: ChooseUser
+
+    user_input = yield ChoseUserForm
+```
+
+To prefill the user_id, you need to add the value to the prop, for prefilling the group_id you need to create a new class:
+
+```python
+def user_choice(group_id: int | None = None) -> type:
+    namespace = {"group_id": group_id}
+    return new_class(
+        "ChooseUserValue", (ChooseUser,), {}, lambda ns: ns.update(namespace)
+    )
+
+
+def initial_input_form_generator(product: UUIDstr, product_name: str) -> FormGenerator:
+    class ChoseUserForm(FormPage):
+        class Config:
+            title = product_name
+
+        user: user_choice("group_id_1") = "user_id_1"
+
+    user_input = yield ChoseUserForm
+```
+
+### auto field loader
+
+The auto field loader is for loading the correct field component in the form.
+it has switches that checks the field type and the field format, where you can add your custom field to.
```

Comment:
checks => check

Maybe stop after the ","

And add: You have to add your new form field here.